src/client.rs

```diff
@@ -911,14 +996,28 @@ impl Spotify {
     /// - offset - the index of the first album to return
     /// - market - Provide this parameter if you want to apply Track Relinking.
     ///
+    /// See [`Spotify::current_user_saved_albums`] for a manually paginated
+    /// version of this.
+    ///
     /// [Reference](https://developer.spotify.com/documentation/web-api/reference/#endpoint-get-users-saved-albums)
+    pub fn current_user_saved_albums(&self) -> impl Paginator<ClientResult<SavedAlbum>> + '_ {
```

Comment:
Yup, otherwise you get this:

```rust
   Compiling rspotify v0.10.0 (/home/mario/Programming/rspotify)
error[E0759]: `self` has an anonymous lifetime `'_` but it needs to satisfy a `'static` lifetime requirem
ent
    --> src/client.rs:1005:13
     |
1003 |     pub fn current_user_saved_albums(&self) -> impl Paginator<ClientResult<SavedAlbum>> {
     |                                      ----- this data with an anonymous lifetime `'_`...
1004 |         paginate(
1005 |             move |limit, offset| self.current_user_saved_albums_manual(Some(limit), Some(offset)),
     |             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
...is captured here...
     |
note: ...and is required to live as long as `'static` here
    --> src/client.rs:1003:48
     |
1003 |     pub fn current_user_saved_albums(&self) -> impl Paginator<ClientResult<SavedAlbum>> {
     |                                                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
help: to declare that the `impl Trait` captures data from argument `self`, you can add an explicit `'_` l
ifetime bound
     |
1003 |     pub fn current_user_saved_albums(&self) -> impl Paginator<ClientResult<SavedAlbum>> + '_ {
     |                                                                                         ^^^^

error[E0759]: `self` has an anonymous lifetime `'_` but it needs to satisfy a `'static` lifetime requirem
ent
    --> src/client.rs:1005:39
     |
1003 |     pub fn current_user_saved_albums(&self) -> impl Paginator<ClientResult<SavedAlbum>> {
     |                                      ----- this data with an anonymous lifetime `'_`...
1004 |         paginate(
1005 |             move |limit, offset| self.current_user_saved_albums_manual(Some(limit), Some(offset)),
     |                                  ---- ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
     |                                  |
     |                                  ...is captured here...
     |
note: ...and is required to live as long as `'static` here
    --> src/client.rs:1003:48
     |
1003 |     pub fn current_user_saved_albums(&self) -> impl Paginator<ClientResult<SavedAlbum>> {
     |                                                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
help: to declare that the `impl Trait` captures data from argument `self`, you can add an explicit `'_` l
ifetime bound
     |
1003 |     pub fn current_user_saved_albums(&self) -> impl Paginator<ClientResult<SavedAlbum>> + '_ {
     |                                                                                         ^^^^

error: aborting due to 2 previous errors

For more information about this error, try `rustc --explain E0759`.
error: could not compile `rspotify`

To learn more, run the command again with --verbose.
```

Reply:
Wait, but I *can* remove it from the `paginate` function. Give me a sec.

Reply:
The `self` has an anonymous lifetime `'_`, in order to explicitly indicate the return value has can live as long as the `self`, so we just simply add an `'_` for the return value? is it right?

Reply:
Yup, the thing is that `paginate` captures `self`, which has a borrowed lifetime of `'_`. Thus, we must indicate that the returned value won't outlive `self` with `'_`. I thought this was done automatically by the compiler but being an `impl` it makes sense that it's forcing us to specify the lifetime.